src/sys/windows/tcp.rs

```diff
@@ -323,6 +328,7 @@ impl Read for TcpStream {
                 // Once the entire buffer is written we need to schedule the
                 // next read operation.
                 if cursor.position() as usize == cursor.get_ref().len() {
+                    me.iocp.unset_readiness(EventSet::readable());
```

Comment:
Perhaps these calls to `unset` can happen in the `schedule_foo` functions? I think that may be a bit more central of a location and it'll also handle the case where the operation completes immediately (e.g. the readiness doesn't need to be unset)